src/tracepoint_format_parser.cpp

```diff
@@ -90,16 +90,9 @@ bool TracepointFormatParser::parse(ast::ASTContext &ctx, BPFtrace &bpftrace)
           if (format_file.fail()) {
             // Errno might get clobbered by LOG().
             int saved_errno = errno;
-
-            // Do not fail if trying to attach to multiple tracepoints
-            // (at least one of them could succeed)
-            bool fail = probe->attach_points.size() == 1;
             auto msg = "tracepoint not found: " + category + ":" + event_name;
             auto select = [&]() -> ast::Diagnostic & {
-              if (fail)
-                return ap->addError();
-              else
-                return ap->addWarning();
+              return ap->addWarning();
```

Comment:
The only thing `select` was doing is basically adding an error or warning diagnostic. Just drop this in favor of `auto &warn = ap->addWarning();`